tests/testthat/test-plot_seromodel.R

```diff
@@ -0,0 +1,313 @@
+library(testthat)
+library(ggplot2)
+library(rlang)
+library(purrr)
+
+# NOTE: If the variable `expected_plot` needs to be updated because of changes in the plots
+# You can use dput(extract_plot_data(plot)) to obtain the current structure of the plot
+
+# Common data
+data(veev2012)
+serosurvey <- veev2012
+
+
+create_prepared_serosurvey <- function(actual_serosurvey) {
+  return(prepare_serosurvey_for_plotting(actual_serosurvey %>%
+    add_age_group_to_serosurvey()))
+}
+
+
+test_that("plot_serosurvey creates a ggplot with correct structure", {
+  skip_on_cran()
+
+  seromodel <- fit_seromodel(
+    serosurvey = serosurvey
+  )
+  actual_plot <- extract_plot_data(plot_serosurvey(serosurvey))
+
+  prepared_serosurvey <- create_prepared_serosurvey(serosurvey)
+
+  expected_plot <- list(
+    classes = c("gg", "ggplot"),
+    layers = list(
+      list(
+        mapping = list(
+          ymin = "seroprev_lower", ymax = "seroprev_upper"
+        ), geom_params = list(
+          na.rm = FALSE, orientation = NA, width = 0.1
+        )
+      ),
+      list(
+        mapping = list(
+          y = "seroprev", size = "n_sample"
+        ), geom_params = list(na.rm = FALSE)
+      )
+    ),
+    coordinates = list(limits = list(x = c(1, 60), y = c(
+      0.3,
+      1
+    ))),
+    labels = list(
+      x = "Age", y = "Seroprevalence", ymin = "seroprev_lower",
+      ymax = "seroprev_upper", size = "n_sample"
+    )
+  )
+
+  expect_lists_equal_with_tolerance(expected_plot, actual_plot)
+})
+
+
+
+test_that("plot_serosurvey creates a binned ggplot with correct structure", {
+  skip_on_cran()
+
+  seromodel <- fit_seromodel(
+    serosurvey = serosurvey
+  )
+  actual_plot <- extract_plot_data(plot_serosurvey(serosurvey, bin_serosurvey = TRUE, bin_step = 10))
+
+  prepared_serosurvey <- create_prepared_serosurvey(serosurvey)
+
+  expected_plot <- list(
+    classes = c("gg", "ggplot"),
+    layers = list(
+      list(
+        mapping = list(
+          ymin = "seroprev_lower", ymax = "seroprev_upper"
+        ),
+        geom_params = list(
+          na.rm = FALSE, orientation = NA, width = 0.1
+        )
+      ),
+      list(
+        mapping = list(
+          y = "seroprev", size = "n_sample"
+        ),
+        geom_params = list(na.rm = FALSE)
+      )
+    ),
+    coordinates = list(
```

Comment:
This change would require to make the test code too complex, since `plot_serosurvey` performs too many changes to `serosurvey`. It's easier to maintain hard-coded values and, if the data changes, one can run `dput(extract_plot_data(actual_plot))` to obtain the current structure of the plot